### ISSUE TITLE
Incorrect L034 detection

### ISSUE THREAD

**Issue by flvndh** [sqlfluff/sqlfluff#4001]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

I have the following CTE

```sql
with my_cte as (
    select
        regexp_extract(upper(column_1), '((PAT|SAT|SBF)\\.[A-Z0-9]{3})', 0) as my_column_1,
        column_2 as my_column_2
    from
        {{ ref('my_table') }}
)

select
    *
from
    my_cte
```
 that triggers the `L034` rule. a.k.a `Select wildcards then simple targets before calculations and aggregates`.

### Expected Behaviour

The rule is not violated

### Observed Behaviour

SQLFluff see a `*` when there is not

### How to reproduce

```sh
sqlfluff lint
```

### Dialect

databricks

### Version

sqlfluff==1.3.2
sqlfluff-templater-dbt==1.3.2

### Configuration

```ini
[sqlfluff]
dialect = databricks
templater = dbt
sql_file_exts = .sql
 # L051? L055?
exclude_rules = L014,L039
verbose = 2

[sqlfluff:rules]
tab_space_size = 4
max_line_length = 120
indent_unit = space
comma_style = trailing
allow_scalar = false
single_table_references = consistent
unquoted_identifiers_policy = all

[sqlfluff:indentation]
indented_joins = false
indented_using_on = false
indented_on_contents = false
indented_ctes = false
template_blocks_indent = false

[sqlfluff:templater]
unwrap_wrapped_queries = true

[sqlfluff:templater:jinja]
apply_dbt_builtins = false

[sqlfluff:templater:dbt]
project_dir = .
profiles_dir = .
profile = iba
target = databricks

[sqlfluff:templater:dbt:context]
environment = sqlfluff
snapshot_date = 9999-12-31

[sqlfluff:rules:L003]
hanging_indents = false

[sqlfluff:rules:L010]
capitalisation_policy = lower

[sqlfluff:rules:L030]
extended_capitalisation_policy = lower

[sqlfluff:rules:L036]
wildcard_policy = multiple

[sqlfluff:rules:L042]
forbid_subquery_in = both

[sqlfluff:rules:L064]
preferred_quoted_literal_style = single_quotes
```

```
target/
dbt_packages/
macros/
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by peterbud** [comment]
With the latest sqlfluff this linting error is not appearing anymore.
